clients/common/nm-secret-agent-simple.c

```diff
@@ -392,7 +392,18 @@ add_vpn_secret_helper (GPtrArray *secrets, NMSettingVpn *s_vpn, const char *name
 		                                            NM_SETTING (s_vpn),
 		                                            name,
 		                                            nm_setting_vpn_get_service_type (s_vpn));
+	} else if (g_str_has_prefix (name, NM_VPN_INTERACTIVE_SECRET_PREFIX)) {
```

Comment:
this is indeed quite ugly. Above you pass `name` to `get_vpn_secret_flags()`, like it was a regular secret which would be in the `s_vpn` dictionary. Do you expect to find name there? Otherwise, you should first check for that. Actually, the entire order is ugly.

Below, `add_vpn_secrets()` calls `add_vpn_secret_helper()` with `p->name`. Good. In that case, `p->name` is clearly a secret in `s_vpn`. But the other caller of `add_vpn_secret_helper()` assumes that the hints name can be found in `s_vpn`? It seems so ugly.